src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -29,7 +29,8 @@ protected override void CleanupTest ()
 				string deviceLog = Path.Combine (outputDir, "logcat-failed.log");
 				string remote = "/data/local/tmp/screenshot.png";
 				RunAdbCommand ($"shell screencap {remote}");
-				RunAdbCommand ($"logcat -d > \"{deviceLog}\"");
+				var output = RunAdbCommand ($"logcat -d");
+				File.WriteAllText (deviceLog, output);
```

Comment:
Since this could be a 4MB string:

https://github.com/xamarin/xamarin-android/blob/c98cb40f35a208c04466bb3fccb3cd25d463781c/build-tools/scripts/TestApks.targets#L94-L100

Do you think it's worth making a `GetLogcatOutput` method that would write each line to a file at a time? Instead of making a giant string?

Reply:
For tests code I don't care much.

If we really want this, we should do it by extending the `BaseTest.RunCommand` to be able to write the output to the file instead of returning the string with the outputs and let it pass the file path through the `RunAdbCommand `.